commerce_coordinator/apps/commercetools/clients.py

```diff
@@ -451,6 +451,54 @@ def create_return_for_order(self, order_id: str, order_version: int, order_line_
                                        err, f"Unable to create return for order {order_id}")
             raise err
 
+    def update_return_payment_state_for_enrollment_code_purchase(
+        self,
+        order_id: str,
+        order_version: int,
+        return_line_item_return_ids: List[str],
+    ) -> Union[CTOrder, None]:
+        """
+        Update paymentState on the LineItemReturnItem attached to the order for enrollment code purchase.
+        Updated by the Order ID (UUID)
+
+        Args:
+            order_id (str): Order ID (UUID)
+            order_version (int): Current version of order
+            return_line_item_return_id (str): LineItemReturnItem ID
+
+        Returns (CTOrder): Updated order object or
+        Raises Exception: Error if update was unsuccessful.
+        """
+
+        try:
+            logger.info(
+                f"[CommercetoolsAPIClient] - Updating payment state for return "
+                f"with ids {return_line_item_return_ids} to '{ReturnPaymentState.NOT_REFUNDED}'."
+            )
+            return_payment_state_actions = []
+            for return_line_item_return_id in return_line_item_return_ids:
+                return_payment_state_actions.append(OrderSetReturnPaymentStateAction(
+                    return_item_id=return_line_item_return_id,
+                    payment_state=ReturnPaymentState.NOT_REFUNDED,
+                ))
+
+            updated_order = self.base_client.orders.update_by_id(
+                id=order_id,
+                version=order_version,
+                actions=return_payment_state_actions
+            )
+            logger.info(f"Successfully updated return payment state to not refunded "
+                        f"for enrollment code purchase - order_id: {order_id}")
+            return updated_order
+        except CommercetoolsError as err:
+            handle_commercetools_error(
+                "[CommercetoolsAPIClient.update_return_"
+                "payment_state_for_enrollment_code_purchase]",
+                err,
+                f"Unable to update ReturnPaymentState of order {order_id}"
+            )
+            raise OpenEdxFilterException(str(err)) from err
```

Comment:
```suggestion
            raise err
```

this is not called inside a filter so we shouldn't we raising this exception. we should raise CommercetoolsError as is so that task can be retried

Reply:
Done